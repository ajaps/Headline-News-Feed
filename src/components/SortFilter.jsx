import React from 'react';
import PropTypes from 'prop-types';
import * as myActions from '../actions/Headlines';

import ArticleStore from '../stores/Articles';

/**
 * Represents the Available sort filter
 */
export default class SortFilter extends React.Component {
  constructor() {
    super();
    this.bySort = this.bySort.bind(this);
  }
/**
 * sends an action to change the preffered sort filter based on user selection
 * @param {object} e The selected sort Filter
 * @return {void}
 */
  bySort(e) {
    const sortBy = e.target.dataset.sort;
    const SourceId = ArticleStore.getSourceId();
    myActions.setApiSortBy(SourceId, sortBy);
  }

  render() {
    const articleSortFilters = this.props.sortFilter;
    let allAvailableSortFilter;

    if (articleSortFilters) {
      allAvailableSortFilter = articleSortFilters.map(eachSortItem =>
        <li className={eachSortItem} key={eachSortItem}>
          <a data-sort={eachSortItem} onClick={this.bySort} ref={eachSortItem}>
            { eachSortItem } </a>
        </li>
      );
    }

    return (
      <div className="container">
          <div className="dropdown pull-right sortByFeature">
            <a aria-expanded="false" aria-haspopup="true"
                className="dropdown-toggle" data-toggle="dropdown" role="button"
            > Sort Articles By
            <span className="caret" /></a>
            <ul className="dropdown-menu">
              {allAvailableSortFilter}
            </ul>
          </div>
      </div>
    );
  }
}

SortFilter.propTypes = {
  sortFilter: PropTypes.array,
};
