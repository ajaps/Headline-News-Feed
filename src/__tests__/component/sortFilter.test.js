import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import SortFilter from '../../components/SortFilter.jsx';

jest.mock('../../stores/Articles', () => ({
  getArticleUrl: {
    sortBy: '',
    source: 'all',
  }
}));
const myMock = jest.fn();
myMock('bySort');

describe('SortFiler component', () => {
  let app;
  const filter = ['top', 'latest'];
  beforeEach(() => {
    app = shallow(<SortFilter sortFilter={filter} />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should have dropdown children equal to length of props', () => {
    expect(app.find('li').length).toEqual(2);
  });
});
