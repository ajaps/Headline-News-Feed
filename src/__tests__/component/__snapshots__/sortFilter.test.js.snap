// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortFilter component should render as expected 1`] = `
<div
  className="container"
>
  <div
    className="dropdown pull-right sortByFeature"
  >
    <a
      aria-expanded="false"
      aria-haspopup="true"
      className="dropdown-toggle"
      data-toggle="dropdown"
      role="button"
    >
       Sort Articles By
      <span
        className="caret"
      />
    </a>
    <ul
      className="dropdown-menu"
    >
      <li
        className="top"
      >
        <a
          data-sort="top"
          onClick={[Function]}
        >
          top
           
        </a>
      </li>
      <li
        className="latest"
      >
        <a
          data-sort="latest"
          onClick={[Function]}
        >
          latest
           
        </a>
      </li>
    </ul>
  </div>
</div>
`;
